File: backend/app/models/talent_check.py
from enum import Enum
from typing import Dict, List, Optional
from pydantic import BaseModel, Field
from app.models.skill import SkillCategoryCode


class SkillGapStatus(str, Enum):
    MET = "met"
    MINOR_GAP = "minor_gap"
    CRITICAL_GAP = "critical_gap"


class ReadinessTier(str, Enum):
    READY = "Ready"
    NEEDS_PREPARATION = "Needs Preparation"
    SIGNIFICANT_GAP = "Significant Gap"


class ItemizedSkillComparison(BaseModel):
    skill_name: str
    category_code: SkillCategoryCode
    required_level: int = Field(..., ge=1, le=10)
    candidate_level: int = Field(..., ge=1, le=10)
    gap: int = Field(..., ge=0)
    status: SkillGapStatus
    evidence: Optional[str] = None


class CategorySkillComparison(BaseModel):
    category_code: SkillCategoryCode
    category_name: str
    required_level_avg: float
    candidate_level_avg: float
    gap_avg: float
    status: SkillGapStatus
    skills: List[ItemizedSkillComparison] = Field(default_factory=list)


class TalentCheckRequest(BaseModel):
    candidate_id: str
    company_id: int
    custom_skill_assessments: Optional[Dict[str, int]] = None


class TalentCheckResponse(BaseModel):
    candidate_id: str
    company_id: int
    company_name: str
    overall_readiness_score: float = Field(..., ge=0, le=100)
    readiness_tier: ReadinessTier
    category_comparisons: List[CategorySkillComparison] = Field(default_factory=list)
    key_strengths: List[str] = Field(default_factory=list)
    priority_gaps: List[str] = Field(default_factory=list)
    evaluated_at: str

File: src/shared/types/talentCheck.ts
/**
 * Talent Check Data Contract
 * Request and Response schemas for evaluating a Candidate against a Company placement profile.
 */

import { SkillCategoryCode } from "./skill";

export type SkillGapStatus = "met" | "minor_gap" | "critical_gap";
export type ReadinessTier = "Ready" | "Needs Preparation" | "Significant Gap";

/**
 * Individual skill level comparison within a category
 */
export interface ItemizedSkillComparison {
  skill_name: string;
  category_code: SkillCategoryCode;
  required_level: number; // 1-10 scale
  candidate_level: number; // 1-10 scale
  gap: number; // max(0, required_level - candidate_level)
  status: SkillGapStatus;
  evidence?: string;
}

/**
 * Category-level aggregated skill comparison
 */
export interface CategorySkillComparison {
  category_code: SkillCategoryCode;
  category_name: string;
  required_level_avg: number;
  candidate_level_avg: number;
  gap_avg: number;
  status: SkillGapStatus;
  skills: ItemizedSkillComparison[];
}

/**
 * Talent Check input request
 */
export interface TalentCheckRequest {
  candidate_id: string;
  company_id: number;
  custom_skill_assessments?: Record<string, number>; // optional manual overrides for self-assessment
}

/**
 * Talent Check evaluation response
 */
export interface TalentCheckResponse {
  candidate_id: string;
  company_id: number;
  company_name: string;
  overall_readiness_score: number; // 0 - 100 percentage
  readiness_tier: ReadinessTier;
  category_comparisons: CategorySkillComparison[];
  key_strengths: string[];
  priority_gaps: string[];
  evaluated_at: string;
}
